lib/fog/compute/ecloud/models/ssh_key.rb

```diff
@@ -9,6 +9,30 @@ class SshKey < Fog::Ecloud::Model
         attribute :other_links, :aliases => :Links
         attribute :default, :aliases => :Default, :type => :boolean
         attribute :finger_print, :aliases => :FingerPrint
+        attribute :private_key, :aliases => :PrivateKey
+
+        def delete
+          service.ssh_key_delete(href).body
+        end
+        alias_method :destroy, :delete
+
+        def edit(options = {})
+          # Make sure we only pass what we should
+          new_options = {}
+          new_options[:uri] = href
+          if options[:name].nil?
+            new_options[:name] = name
+          else
+            new_options[:name] = options[:name]
+          end
+          if options[:default].nil?
+            new_options[:default] = default
+          else
+            new_options[:default] = options[:default]
+          end
```

Comment:
That's what it currently accepts, which may change in the future. The "default" field has already been deprecated, and could be removed at a later date.

The nil checking is because the XML request requires the "name" attribute, regardless of whether it's changing or not. So, basically, we're creating a fully formed dataset using current values when they're not supplied.